mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,47 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last and 'last_ckpt' in self.runtime_info:
```

Comment:
where is `self.runtime_info` defined.

Reply:
`if self.save_last and 'last_ckpt' in runner.message_hub.runtime_info:`